pkg/mdlsub/fixtures.go

```diff
@@ -0,0 +1,192 @@
+		httpClient: httpClient,
+		settings:   settings,
+	}
+}
+
+func (f FixtureSet) Write(ctx context.Context) error {
+	var ok bool
+	var err error
+	var res *FetchResult
+	var transformer ModelTransformer
+	var unmarshal func(data []byte, v interface{}) error
+	var output Output
+	var items []interface{}
+	var model Model
+
+	if res, err = f.Fetch(ctx); err != nil {
+		return fmt.Errorf("failed to fetch model %s: %w", f.source.String(), err)
+	}
+
+	if transformer, err = f.core.GetTransformer(res.spec); err != nil {
+		return fmt.Errorf("failed to get transformer: %w", err)
+	}
+
+	input := transformer.GetInput()
+	slice := refl.CreatePointerToSliceOfTypeAndSize(input, 0)
+
+	if unmarshal, ok = unmarhaller[res.contentType]; !ok {
+		return fmt.Errorf("unknown content type: %s", res.contentType)
+	}
+
+	if err = unmarshal(res.data, slice); err != nil {
+		return fmt.Errorf("failed to unmarshal fetched data: %w", err)
+	}
+
+	if items, err = refl.InterfaceToInterfaceSlice(slice); err != nil {
+		return fmt.Errorf("failed to interface slice: %w", err)
+	}
+
+	if output, err = f.core.GetOutput(res.spec); err != nil {
+		return fmt.Errorf("failed to get output: %w", err)
+	}
+
+	for _, item := range items {
+		mdl := refl.ValueToPointerValue(item)
+
+		if model, err = transformer.Transform(ctx, mdl); err != nil {
+			return fmt.Errorf("failed to transform model %s: %w", f.source.String(), err)
+		}
+
+		if err = output.Persist(ctx, model, res.spec.CrudType); err != nil {
+			return fmt.Errorf("failed to persist model %s: %w", f.source.String(), err)
+		}
+	}
+
+	f.logger.Info("%s: persisted %d fixtures", f.source.String(), len(items))
+
+	return nil
+}
+
+func (f FixtureSet) Fetch(ctx context.Context) (*FetchResult, error) {
+	var err error
+	var hostPath string
+	var u *url.URL
+	var resp *resty.Response
+	var versionInt int
+
+	if hostPath, err = url.JoinPath(f.settings.Host, f.settings.Path); err != nil {
+		return nil, fmt.Errorf("failed to join host path: %w", err)
+	}
+
+	if u, err = url.Parse(hostPath); err != nil {
+		return nil, fmt.Errorf("failed to parse host url %s: %w", hostPath, err)
+	}
+
+	query := u.Query()
+	query.Add("data_set", f.settings.DataSetDbName)
+	query.Add("model_id", f.source.String())
+	u.RawQuery = query.Encode()
+
+	req := f.httpClient.R().SetContext(ctx)
+	f.logger.Info("%s: fetching fixture data from %s", f.source.String(), u.String())
+
+	if resp, err = req.Get(u.String()); err != nil {
+		return nil, fmt.Errorf("error on executing http request data: %w", err)
+	}
+
+	if resp.StatusCode() != http.StatusOK {
+		return nil, fmt.Errorf("error on executing http request data, got status %d: %s", resp.StatusCode(), resp.String())
+	}
+
+	version := resp.Header().Get("Version")
+
+	if version == "" {
+		return nil, fmt.Errorf("version not found in response header")
+	}
+
+	if versionInt, err = strconv.Atoi(version); err != nil {
+		return nil, fmt.Errorf("can not convert version string %s to int: %w", version, err)
+	}
```

Comment:
My current codes expectation is that you supply the version you want to receive, as i need that in the transformers. How should we resolve this?

Reply:
My suggestion here would be to add a `GetModelIdVersions(modelId string) []int` func or a `GetLatestModelIdVersion(modelId string) int` func to the subscriber core, which could be used within the Fetch function to let it query the latest version available.

Reply:
done

Reply:
done